functions/New-DbaDbTransfer.ps1

```diff
@@ -0,0 +1,250 @@
+        License: MIT https://opensource.org/licenses/MIT
+
+    .LINK
+        https://dbatools.io/New-DbaDbTransfer
+
+    .EXAMPLE
+        PS C:\> New-DbaDbTransfer -SqlInstance sql1 -Destination sql2 -Database mydb -CopyAll Tables
+
+        Creates a transfer object that, when invoked, would copy all tables from database sql1.mydb to sql2.mydb
+
+    .EXAMPLE
+        PS C:\> Get-DbaDbTable -SqlInstance sql1 -Database MyDb -Table a, b, c | New-DbaDbTransfer -SqlInstance sql1 -Destination sql2 -Database mydb
+
+        Creates a transfer object to copy specific tables from database sql1.mydb to sql2.mydb
+    #>
+    [OutputType([Microsoft.SqlServer.Management.Smo.Transfer])]
+    [CmdletBinding(DefaultParameterSetName = "Default")]
+    Param (
+        [Parameter(Mandatory)]
+        [DbaInstanceParameter]$SqlInstance,
+
+        [PSCredential]$SqlCredential,
+
+        [DbaInstanceParameter]$DestinationSqlInstance,
+
+        [PSCredential]$DestinationSqlCredential,
+
+        [Parameter(Mandatory)]
+        [string]$Database,
+
+        [string]$DestinationDatabase = $Database,
+
+        [int]$BatchSize = 50000,
+
+        [int]$BulkCopyTimeOut = 5000,
+
+        [Microsoft.SqlServer.Management.Smo.ScriptingOptions]$ScriptingOption,
+
+        [Parameter(ValueFromPipeline)]
+        [Microsoft.SqlServer.Management.Smo.NamedSmoObject[]]$InputObject,
+
+        [switch]$CopyAllObjects,
+
+        [ValidateSet(
+            'FullTextCatalogs',
+            'FullTextStopLists',
+            'SearchPropertyLists',
+            'Tables',
+            'Views',
+            'StoredProcedures',
+            'UserDefinedFunctions',
+            'UserDefinedDataTypes',
+            'UserDefinedTableTypes',
+            'PlanGuides',
+            'Rules',
+            'Defaults',
+            'Users',
+            'Roles',
+            'PartitionSchemes',
+            'PartitionFunctions',
+            'XmlSchemaCollections',
+            'SqlAssemblies',
+            'UserDefinedAggregates',
+            'UserDefinedTypes',
+            'Schemas',
+            'Synonyms',
+            'Sequences',
+            'DatabaseTriggers',
+            'DatabaseScopedCredentials',
+            'ExternalFileFormats',
+            'ExternalDataSources',
+            'Logins',
+            'ExternalLibraries'
+        )]
+        [string[]]$CopyAll,
+
+        [switch]$SchemaOnly,
+
+        [switch]$DataOnly,
+
+        [switch]$EnableException
+    )
+    begin {
+        $objectCollection = New-Object System.Collections.ArrayList
+    }
+    process {
+        foreach ($object in $InputObject) {
+            if (-not $object) {
+                Stop-Function -Message "Object is empty"
+                return
+            }
+            $objectCollection.Add($object) | Out-Null
+        }
+
+    }
+    end {
+        $sourceDb = Get-DbaDatabase -SqlInstance $SqlInstance -SqlCredential $SqlCredential -Database $Database -EnableException:$EnableException
```

Comment:
You don't need to map EnableException with the call if the command in the base function.